R/bookmark-state.R

```diff
@@ -76,7 +76,25 @@ saveShinySaveState <- function(state) {
 
   # Pass the saveState function to the save interface function, which will
   # invoke saveState after preparing the directory.
-  saveInterface <- getShinyOption("save.interface", default = saveInterfaceLocal)
+
+  # Look for a save.interface function. This will be defined by the hosting
+  # environment if it supports bookmarking.
+  saveInterface <- getShinyOption("save.interface")
+
+  if (is.null(saveInterface)) {
+    if (inShinyServer()) {
+      # We're in a version of Shiny Server/Connect that doesn't have
+      # bookmarking support.
+      saveInterface <- function(id, callback) {
+        stop("The hosting environment does not support saved-to-server bookmarking.")
```

Comment:
This waits until someone actually attempts to save a bookmark to error. Can we also give an error when the app starts, or something?